modules/payment/src/Plugin/Commerce/PaymentGateway/OffsitePaymentGatewayInterface.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Drupal\commerce_payment\Plugin\Commerce\PaymentGateway;
+
+use Drupal\commerce_order\Entity\OrderInterface;
+
+/**
+ * Defines the base interface for off-site payment gateways.
+ */
+interface OffsitePaymentGatewayInterface extends PaymentGatewayInterface {
+
+  /**
+   * Gets the URL for offiste redirect cancel.
+   *
+   * @param \Drupal\commerce_order\Entity\OrderInterface $order
+   *   The order.
+   *
+   * @return \Drupal\Core\Url
+   *   The Url object
+   */
+  public function getPaymentRedirectCancelUrl(OrderInterface $order);
+
+  /**
+   * Gets the URL for offiste redirect return.
+   *
+   * @param \Drupal\commerce_order\Entity\OrderInterface $order
+   *   The order.
+   *
+   * @return \Drupal\Core\Url
+   *   The Url object
+   */
+  public function getPaymentRedirectReturnUrl(OrderInterface $order);
+
+  /**
+   * Invoked when the off-site payment return.
+   *
+   * @param \Drupal\commerce_order\Entity\OrderInterface $order
+   *   The order.
+   */
+  public function onRedirectReturn(OrderInterface $order);
```

Comment:
Maybe we just remove Payment from all of then? getRedirectReturnUrl() getRedirectCancelUrl(), onRedirectReturn() onRedirectCancel()? The return/cancel duality is confusing, cause technically both urls are return urls, one is for success one is for failure. Maybe we do SuccessUrl and CancelUrl() instead?